lib/components/app/responsive-webapp.js

```diff
@@ -31,34 +29,69 @@ class ResponsiveWebapp extends Component {
 
   /** Lifecycle methods **/
 
-  componentWillReceiveProps (nextProps) {
-    const {query} = this.props
+  componentDidUpdate (prevProps) {
+    const { currentPosition, location, query, title } = this.props
+    document.title = title
 
-    if (!isEqual(query, nextProps.query)) {
-      this.props.formChanged(query, nextProps.query)
+    if (!isEqual(prevProps.query, query)) {
+      // console.log('form changed', prevProps.query, query)
+      this.props.formChanged(prevProps.query, query)
+    } else {
+      // If the query does not change, but the search has, this likely means the
+      // user has pressed the back button, and we need to update the query from
+      // the search string and re-perform the search.
+      const urlParams = getUrlParams()
+      const planParams = planParamsToQuery(urlParams)
+      let hasChangedParam = false
+      Object.keys(planParams)
+        .forEach(key => {
+          let paramIsDifferent = false
+          switch (key) {
+            case 'from':
+            case 'to':
+              if (!matchLatLon(planParams[key], query[key])) {
+                paramIsDifferent = true
+              }
+              break
+            case '':
+            default:
+              if (!isEqual(planParams[key], query[key])) {
+                paramIsDifferent = true
+              }
+              break
+          }
+          if (paramIsDifferent) {
+            hasChangedParam = true
+          }
+        })
+      if (hasChangedParam) {
+        console.log('handle search change', prevProps.location.search, location.search)
+        console.log(query, planParams)
+        // this.props.setQueryParam(planParams)
```

Comment:
What does `hasChangedParam` do? Is it used anymore?

Reply:
Ah, forgot to finish this up. It's supposed to basically handle when the back button is pressed, detecting the URL search change and resending the trip plan request.

Reply:
OK, the back button behavior has been fully implemented.